apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PriorityList.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// A simple sorted list that allows items to return equality for IComparable.CompareTo
+    /// </summary>
+    internal class PriorityList<T> where T : IPriorityListItem<T>
+    {
+        private readonly List<T> _innerList = new List<T>();
+
+        public int Count
+        {
+            get
+            {
+                lock (_innerList)
+                {
+                    return _innerList.Count;
+                }
+            }
+        }
+
+        public IEnumerable<T> GetSnapshot()
+        {
+            lock (_innerList)
+            {
+                return _innerList.ToArray();
+            }
+        }
+
+        public void Add(T item)
```

Comment:
Possibly rename to AddIfAbsent? I got confused reading later code that looked like it was unconditionally adding an item that might already be in the list.